internal/pkg/diff/diff.go

```diff
@@ -18,13 +18,26 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/strhelper"
 )
 
+type Option func(c *Config)
+
+type Config struct {
+	allowTargetEnv bool
+}
+
+func WithAllowTargetEnv(allowTargetEnv bool) Option {
+	return func(c *Config) {
+		c.allowTargetEnv = allowTargetEnv
+	}
+}
+
 type typeName string
 
 type Differ struct {
-	objects   model.ObjectStates
-	results   []*Result                                 // diff results
-	typeCache map[typeName][]*reflecthelper.StructField // reflection cache
-	errors    errors.MultiError
+	objects        model.ObjectStates
+	results        []*Result                                 // diff results
+	typeCache      map[typeName][]*reflecthelper.StructField // reflection cache
+	allowTargetEnv bool
```

Comment:
Put it as last in struct, so it is obvious that it is an options for `Differ`. Maybe add commentary `// option` ?